doc/guides: Add clarifications to board porting guidelines

erwango opened the pull request:
Additions:
- Provide clear description of a typical board port on zephyr
- Add a clear statement that peripherals should be disabled by
default (unless clearly specified)
- Add clear mentions on peripheral that should actually be enabled

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ioannisg commented on doc/guides/porting/board_porting.rst:
```diff
@@ -120,6 +120,14 @@ Board
 A board implements an SoC with all its features, together with peripherals
 available on the board that differentiates the board with additional interfaces
 and features not available in the SoC.
+A board port on Zephyr is typically made of two parts:
```
typically consists of two parts

ioannisg commented on doc/guides/porting/board_porting.rst:
```diff
@@ -120,6 +120,14 @@ Board
+  buttons, sensors or communication peripheral (USB, BLE controller, ..)
+
+- A software configuration (done using Kconfig) which configures and optionally
+  enables software features that are needed to operate the board.
```
Doesn't the same apply for device tree? I.e. we need to configure and enable things needed to properly operate the board?  E.g. uart pins, enabling peripherals etc.

erwango replied:
Agree it is not clear. Can you check the proposal made to Anas below ?

ioannisg reviewed (commented):
Some minor suggestions from my side

nashif commented on doc/guides/porting/board_porting.rst:
```diff
@@ -120,6 +120,14 @@ Board
+A board port on Zephyr is typically made of two parts:
+
+- A hardware description (usually done by device tree), which specifies embedded
+  SoC reference, connectors and any other hardware components sch as LEDs,
```
s/sch/such

nashif commented on doc/guides/porting/board_porting.rst:
```diff
@@ -120,6 +120,14 @@ Board
+
+- A hardware description (usually done by device tree), which specifies embedded
+  SoC reference, connectors and any other hardware components sch as LEDs,
+  buttons, sensors or communication peripheral (USB, BLE controller, ..)
```
s/peripheral/peripherals/

nashif commented on doc/guides/porting/board_porting.rst:
```diff
@@ -120,6 +120,14 @@ Board
+  buttons, sensors or communication peripheral (USB, BLE controller, ..)
+
+- A software configuration (done using Kconfig) which configures and optionally
+  enables software features that are needed to operate the board.
```
software features are not necessarily needed to "operate the board", maybe "needed by the application"?

erwango replied:
Point was to highlight that features to be enabled are the minimal set required by all applications. But reading it again, agree this is not clear
What about the following ?
```
A software configuration (done using Kconfig) of features and peripheral drivers. This configuration should enable the minimal set common to every application to operate the board.
```

ioannisg replied:
It's not really the minimal @erwango , e.g. we enable ARM MPU in almost all arm boards, but that is not needed for the boards to work, neither by several applications. I think we just select a default set of components.

"A boot time configuration (via Kconfig) which selects and configures the behavior of 
software components (kernel, peripheral drivers, subsystems etc.) enabled by default on this board."

erwango replied:
> It's not really the minimal @erwango [...] I think we just select a default set of components.

Well, initial point of "default board configuration guidelines" was actually to define what is application responsibility and what could be directly enabled on boards (which, as a consequence, is the minimal set required by all applications).
So, this is meant to actually give guidelines here and not only documenting what is a board configuration.

> e.g. we enable ARM MPU in almost all arm boards, but that is not needed for the boards to work, neither by several applications.

I indeed hesitated in giving directions here. Ideally we should. Do we consider this should be part of default configuration (either enabled or disabled)?
(If we decide there should be a direction, this should be done in another PR as we are only clarifying the existing here)

erwango replied:
@ioannisg, I've pushed fixes for all comments done including proposal on this one.
Please confirm if you agree on my point above or if point is still open.

mbolivar commented on doc/guides/porting/board_porting.rst:
```diff
@@ -137,7 +145,10 @@ overridden.
 In order to provide consistency across the various boards and ease the work of
 users providing applications that are not board specific, the following
-guidelines should be followed when porting a board:
+guidelines should be followed when porting a board. Unless explicitly stated
```
```suggestion
guidelines should be followed when porting a board. Unless explicitly stated,
```

mbolivar commented on doc/guides/porting/board_porting.rst:
```diff
@@ -149,12 +160,14 @@ guidelines should be followed when porting a board:
 
 - Enable all GPIO ports.
 
+- If available, enable pinmux and interrupt controller drvers.
```
```suggestion
- If available, enable pinmux and interrupt controller drivers.
```

mbolivar commented on doc/guides/porting/board_porting.rst:
```diff
@@ -149,12 +160,14 @@ guidelines should be followed when porting a board:
   configuring an SPI instance for Arduino SPI.
 
-- Configure an output for the console.
+- Configure and enable a serial output for the console.
```
```suggestion
- If available, configure and enable a serial output for the console.
```

dbkinder commented on doc/guides/porting/board_porting.rst:
```diff
@@ -120,6 +120,16 @@ Board
 A board implements an SoC with all its features, together with peripherals
 available on the board that differentiates the board with additional interfaces
 and features not available in the SoC.
+A board port on Zephyr typically consists of two parts:
```
This feels like the beginning of a new paragraph, so add a blank line before it.

dbkinder commented on doc/guides/porting/board_porting.rst:
```diff
@@ -120,6 +120,16 @@ Board
+  buttons, sensors or communication peripherals (USB, BLE controller, ..)
+
+- A software configuration (done using Kconfig) of features and peripheral
+  drivers. This configuration should enable the minimal set common to every
```
How about:
```
This default board configuration should enable a minimal set of features
common to many applications and to applicable project-provided
:ref:`samples-and-demos`.
```
This also acts as a lead-in to the next section.

carlescufi reviewed (approved):
LGTM!

erwango committed:
doc/guides: Add clarifications to board porting guidelines

Additions:
- Provide clear description of a typical board port on zephyr
- Add a clear statement that peripherals should be disabled by
default (unless clearly specified)
- Add clear mentions on peripheral that should actually be enabled

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
